x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -521,8 +524,19 @@ private Expression onlyOnePipeArgument(Source source, String pipeName, List<Bool
 
     private Expression pipeIntArgument(Source source, String pipeName, List<BooleanExpressionContext> exps) {
         Expression expression = onlyOnePipeArgument(source, pipeName, exps);
+        boolean foldableInt = expression.dataType().isInteger() && expression.foldable();
+        boolean foldingError = false;
+        Number value = null;
+
+        if (foldableInt) {
+            try {
+                value = (Number) expression.fold();
+            } catch (ArithmeticException ae) {
+                foldingError = true;
```

Comment:
Should we simply check `value` against `null` later, rather then setting a flag here, in case folding yields a `null` w/o an exception?

Reply:
Yes. Providing a `null` as a value to `head` or `tail` still breaks the "positive integer" restriction.